### PR TITLE
[beeswax] adding config flag for max no of tables in the left assist

### PR DESCRIPTION
**Pull request by agl29**
## What changes were proposed in this pull request?

- adding config flag for max_tables in the left assit

### CODE REVIEW

File: apps/beeswax/src/beeswax/server/hive_server2_lib.py [hunk 1]
```diff
@@ -869,7 +869,7 @@ def get_tables(self, database, table_names, table_types=None):
     req = TGetTablesReq(schemaName=database, tableName=table_names, tableTypes=table_types)
     (res, session) = self.call(self._client.GetTables, req)
 
-    results, schema = self.fetch_result(res.operationHandle, orientation=TFetchOrientation.FETCH_NEXT, max_rows=5000)
+    results, schema = self.fetch_result(res.operationHandle, orientation=TFetchOrientation.FETCH_NEXT, max_rows=MAX_TABLES_IN_LEFT_ASSIST.get())
```
Comment (JohanAhlen) [inline apps/beeswax/src/beeswax/server/hive_server2_lib.py:872]: The limit is in more places than here, just search the code for "=5000"

File: apps/beeswax/src/beeswax/conf.py [hunk 2]
```diff
@@ -358,3 +358,10 @@ def has_multiple_sessions():
 
 def has_session_pool():
   return has_multiple_sessions() and not CLOSE_SESSIONS.get()
+
+MAX_TABLES_IN_LEFT_ASSIST = Config(
```
Comment (JohanAhlen) [inline apps/beeswax/src/beeswax/conf.py:362]: It's also for DBs and I guess columns, and it's throughout the UI, autocomplete, table browser etc. Perhaps MAX_UI_SQL_ENTRIES is more suitable?

It'd also be good to add something on the lines of "Setting this higher than the default can degrade performance."

File: desktop/conf/pseudo-distributed.ini.tmpl [hunk 3]
```diff
@@ -1369,6 +1369,9 @@
   # Enable the HPLSQL mode.
   ## hplsql=false
 
+  # Max number of tables available to list in the left assist.
```
Comment (JohanAhlen) [inline desktop/conf/pseudo-distributed.ini.tmpl:1372]: Add the performance related comment here as well.

File: apps/beeswax/src/beeswax/conf.py (hunk 2 above)
Comment (agl29) [reply apps/beeswax/src/beeswax/conf.py:362]: okay, got it.

File: apps/beeswax/src/beeswax/server/hive_server2_lib.py (hunk 1 above)
Comment (JohanAhlen) [reply apps/beeswax/src/beeswax/server/hive_server2_lib.py:872]: Go for the same flag.

**Comment by agl29** [comment]
@JohanAhlen  we are using rows=5000,1000,100 in this file also https://github.com/cloudera/hue/blob/master/apps/beeswax/src/beeswax/server/dbms.py
so need to change only 5000 one or all or not needed here?

**Review by JohanAhlen — APPROVED** [review]

**Pull request merged**